python/semantic_kernel/filters/prompt/prompt_render_context.py

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from semantic_kernel.filters.filter_context_base import FilterContextBase
+
+
+class PromptRenderContext(FilterContextBase):
+    """Context for prompt rendering filters."""
+
+    rendered_prompt: str | None = None
```

Comment:
During implementation in .NET, I noticed that it's useful to have function result object in prompt render context as well:
https://github.com/microsoft/semantic-kernel/blob/0c89e0bd4314b4f4c913563258ffefadedab1afe/dotnet/src/SemanticKernel.Abstractions/Filters/Prompt/PromptRenderContext.cs#L66-L70

It will be empty for users, but it's possible to set it during prompt rendering operation, which allows you to avoid function invocation and return result immediately. 

I found it useful for semantic caching scenarios, when you have rendered prompt in prompt filter, you check cache if this prompt was already answered, and if yes - you return result immediately:
https://github.com/microsoft/semantic-kernel/blob/0c89e0bd4314b4f4c913563258ffefadedab1afe/dotnet/samples/Concepts/Caching/SemanticCachingWithFilters.cs#L177-L192

Reply:
very nice, added, will add a sample for that later